File: python/intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        current = intervals[0]
        finalanswer = []
        for i in range(1, len(intervals)):
            if intervals[i][0] <= current[1]:
                current[1] = max(current[1], intervals[i][1])
            else:
                finalanswer.append(current)
                current = intervals[i]

        if current not in finalanswer:
            finalanswer.append(current)

        return finalanswer

File: c++/intervals/merge-intervals.cc
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        std::vector<std::vector<int>> combined_intervals = intervals;        
        std::sort(combined_intervals.begin(), combined_intervals.end());

        std::vector<std::vector<int>> final_intervals = {combined_intervals[0]};
        for (int i = 1; i < combined_intervals.size(); ++i) {
            if (combined_intervals[i][0] <= final_intervals[final_intervals.size() - 1][1]) {
                final_intervals[final_intervals.size() - 1][1] = std::max(final_intervals[final_intervals.size() - 1][1], combined_intervals[i][1]);
            } else {
                final_intervals.push_back(combined_intervals[i]);
            }
        }

        return final_intervals;
    }
};
